pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -489,11 +489,22 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, data *cloudprovidertypes.P
 	_, err = p.get(machine)
 	// If a defunct VM got created, the `Get` call returns an error - But not because the request
 	// failed but because the VM has an invalid config hence always delete except on err == cloudprovidererrors.ErrInstanceNotFound
-	if err != cloudprovidererrors.ErrInstanceNotFound {
-		klog.Infof("deleting VM %q", machine.Name)
-		if err = deleteVMsByMachineUID(context.TODO(), config, machine.UID); err != nil {
-			return false, fmt.Errorf("failed to delete instance for  machine %q: %v", machine.Name, err)
+	if err != nil {
+		if err == cloudprovidererrors.ErrInstanceNotFound {
```

Comment:
this "if err cloudprovidererrors.ErrInstanceNotFound then remove finalizer" is getting repetitive, should this be extracted somehow?

Reply:
maybe we can add util functionality to check for that instead of keep writing all the time. WDYT ?

Reply:
yeah, something like `DeleteFinalizerOnInstanceNotFound(error, *v1alpha1.Machine) (bool, error) {}`

Reply:
Sounds like a good idea. Plus we need a `data *github.com/kubermatic/machine-controller/pkg/cloudprovider/types.ProviderData`  as a third parameter to execute the update. 

Since this is out of the scope for this PR I would like to create a ticket for it and do it separately, because all the current provider is doing the same currently and I don't want to touch all of them in this PR